lib/proxy/log_buffer/decorator.rb

```diff
@@ -52,30 +53,68 @@ def add(severity, message = nil, progname = nil, backtrace = nil)
       end
     end
 
-    def debug(msg_or_progname, exception_or_backtrace = nil, &block)
+    def trace?
+      @trace ||= !!ENV['FOREMAN_TRACE']
+    end
+
+    def trace(msg_or_progname = nil, exception_or_backtrace = nil, &block)
+      add(::Logger::Severity::DEBUG, nil, msg_or_progname, exception_or_backtrace, &block) if trace?
+    end
+
+    def debug(msg_or_progname = nil, exception_or_backtrace = nil, &block)
       add(::Logger::Severity::DEBUG, nil, msg_or_progname, exception_or_backtrace, &block)
     end
 
-    def info(msg_or_progname, exception_or_backtrace = nil, &block)
+    def info(msg_or_progname = nil, exception_or_backtrace = nil, &block)
```

Comment:
You cannot "skip" the first argument so I'm not sure that it can solve anything by adding `nil` to the first element.

Reply:
The example you provided is for "lazy" logging, that is, it is not written right away, and if some of the information was changed when the proc/block were executed, it will hold the new content.

Another way is to do something like this:

```ruby
logger.info "Started #{env['REQUEST_METHOD']} #{env['PATH_INFO']} #{env['QUERY_STRING']}"
logger.trace 'Headers: ' + env.select {|k,v| k.start_with? 'HTTP_'}.inspect 
logger.trace (body = env['rack.input'].read).size > 0 ? 'Body: ' + body : '' 
```

It is logging things right away, and even if the variable changes, it already logged things with the content that was used when executed at that time.

My exemple. uses arguments, rather then the block call, so each argument have it's own place.
If I need to access the second argument, but not the first, Ruby provides a different way of doing so:

```ruby
def info(msg_or_progname: = nil, exception_or_backtrace: = nil, &block)
# ...
end

logger.info exception_or_backtrace: e
```

So the message or progname will be kept as `nil`. 
If you are using it the way your PR does it, your code will be like so:

```ruby
logger.info nil, e
```

Does that make sense?

Reply:
>Does that make sense?

It does not.

Please be aware that proxy runs on Ruby 2.0 in production, I cannot use anything from modern Rubies at the moment, until we SCL it.